pkg/api/v20160330/types.go

```diff
@@ -98,6 +98,31 @@ const (
 // OrchestratorProfile contains Orchestrator properties
 type OrchestratorProfile struct {
 	OrchestratorType OrchestratorType `json:"orchestratorType"`
+	KubernetesConfig KubernetesConfig `json:"kubernetesConfig,omitempty"`
+	DCOSConfig       DCOSConfig       `json:"dcosConfig,omitempty"`
+}
+
+type DCOSConfig struct {
+	DCOS173_BootstrapDownloadURL string
+	DCOS184_BootstrapDownloadURL string
+	DCOS187_BootstrapDownloadURL string
+}
+
+// KubernetesConfig contains the Kubernetes config structure, containing
+// Kubernetes specific configuration
+type KubernetesConfig struct {
+	KubernetesHyperkubeSpec         string `json:"kubernetesHyperkubeSpec,omitempty"`
+	KubectlVersion                  string `json:"kubectlVersion,omitempty"`
+	KubernetesHeapsterSpec          string `json:"kubernetesHeapsterSpec,omitempty"`
+	KubernetesAddonResizerSpec      string `json:"kubernetesAddonResizerSpec,omitempty"`
+	KubernetesKubeDNSSpec           string `json:"kubernetesKubeDNSSpec,omitempty"`
+	KubernetesExecHealthzSpec       string `json:"kubernetesExecHealthzSpec,omitempty"`
+	KubernetesDashboardSpec         string `json:"kubernetesDashboardSpec,omitempty"`
+	KubernetesAddonManagerSpec      string `json:"kubernetesAddonManagerSpec,omitempty"`
+	KubernetesDNSMasqSpec           string `json:"kubernetesDNSMasqSpec,omitempty"`
+	KubernetesPodInfraContainerSpec string `json:"kubernetesPodInfraContainerSpec,omitempty"`
+	KubectlDownloadURL              string `json:"kubectlDownloadURL,omitempty"`
+	DockerInstallScriptURL          string `json:"dockerInstallScriptURL,omitempty"`
 }
```

Comment:
Is this something that really needs to be set in the API model? 

- Once ACS Engine knows the location/environment it can keep a list of download URLs for each location? Or is the expectation that call customer provides this info? If yes, why? How does this work in the K8s scenario today for Azure Public Cloud?

- If there is a need for such a provision providing flexibility to specify a different URL for each spec makes this complicated. Caller should provide a storage account/container/locations and files in that container should be expected to be named in a format that ACS Engine expects. This way we won't required 12 new fields.